datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaMirrorMakerConnector.java

```diff
@@ -575,6 +583,117 @@ public void testReduceDatastreamStates() {
     connector.stop();
   }
 
+  @Test
+  public void testFetchPartitionChange() throws Exception {
+    String clusterName = "testGroupIdAssignment";
+    Properties config = getDefaultConfig(Optional.empty());
+    config.put(AbstractKafkaConnector.IS_GROUP_ID_HASHING_ENABLED, Boolean.toString(true));
+    config.put(KafkaMirrorMakerConnector.PARTITION_FETCH_INTERVAL, "1000");
+
+    KafkaMirrorMakerConnector connector = new KafkaMirrorMakerConnector("MirrorMakerConnector", config, clusterName);
+
+    String yummyTopic = "YummyPizza";
+
+    createTopic(_zkUtils, yummyTopic, 1);
+
+    // create a datastream to consume from topics ending in "Pizza"
+    Datastream datastream = KafkaMirrorMakerConnectorTestUtils.createDatastream("pizzaStream", _broker, "\\w+Pizza");
+    DatastreamGroup group = new DatastreamGroup(ImmutableList.of(datastream));
+    List<DatastreamGroup> datastreamGroups1 = new ArrayList<>();
+    datastreamGroups1.add(group);
+
+    AtomicInteger partitionChangeCalls = new AtomicInteger(0);
+    connector.onPartitionChange(groupName -> {
+      if (groupName.equals(group.getTaskPrefix())) {
```

Comment:
This is certainly a bug since the type of groupName is DatastreamGroup and the type of group.getTaskPrefix() is String.

Reply:
Thanks for the catch, I missed this from refactor.